src/biz/tugay/groovyship/modal/Ship.groovy

```diff
@@ -0,0 +1,20 @@
+package biz.tugay.groovyship.modal
+
+/**
+ * A Ship object is represented as a Map of Coordinate:Boolean where the Boolean value represents whether
+ * the ship coordinate has been hit by a missile or not.
+ * An example might be: [
+ *   [0,0]:false,
+ *   [1,0]:false,
+ *   [2,0]:true
+ * ].
+ * That would be a ship floating horizontally on the first row of a board, where its last piece has been damaged.
+ */
+class Ship
+{
+  Map<Coordinate, Boolean> coordinateIsHitByMissileMap = [:]
```

Comment:
I am open to suggestion about this.

Reply:
I think it would be a list of dictonaries:
eg dictonary x:1, y:2,hit:False or a list of list dictionary of rownumer and list of dictionary:
{x: 1, y:2} : {hit: False} 
There are many possibilities. I would  most likely have a conversion at least to sth like Table has n rows and m columns, either rows or , columns or both contain the content stored in a dictionary or in a Cell class.

Reply:
I initially had a `ShipCoordinate` class, encapsulating a `Coordinate` object and a `boolean isHit` field.. But this approach turned out to be much more simpler..